src/plugins/compressors/sz_threadsafe_plugin.cc

```diff
@@ -0,0 +1,424 @@
+    set(options, "sz_threadsafe:exafel:calib_panel", "for ROIBIN-SZ the size of the calibration panel");
+    set(options, "sz_threadsafe:exafel:num_peaks", "for ROIBIN-SZ the number of peaks");
+    set(options, "sz_threadsafe:exafel:peak_size", "for ROIBIN-SZ the size of the peaks");
+    set(options, "sz_threadsafe:exafel:peaks_cols", "for ROIBIN-SZ the list of columns peaks appear in");
+    set(options, "sz_threadsafe:exafel:peaks_rows", "for ROIBIN-SZ the list of rows peaks appear in");
+    set(options, "sz_threadsafe:exafel:peaks_segs", "for ROIBIN-SZ the segments peaks appear in");
+    set(options, "sz_threadsafe:exafel:sz_dim", "for ROIBIN-SZ the SZ dimensionality prefered");
+    set(options, "sz_threadsafe:exafel:tolerance", "for ROIBIN-SZ the tolerance used after binning");
+    set(options, "sz_threadsafe:gzip_mode", "Which mode to pass to GZIP when used");
+    set(options, "sz_threadsafe:lossless_compressor", "Which lossless compressor to use for stage 4");
+    set(options, "sz_threadsafe:max_quant_intervals", "the maximum number of quantization intervals");
+    set(options, "sz_threadsafe:pred_threshold", "an internal option used to control compression");
+    set(options, "sz_threadsafe:prediction_mode", "an internal option used to control compression");
+    set(options, "sz_threadsafe:psnr_err_bound", "the bound on the error in the PSNR");
+    set(options, "sz_threadsafe:pw_rel_err_bound", "the bound on the pointwise relative error");
+    set(options, "sz_threadsafe:quantization_intervals", "the number of quantization intervals to use, 0 means automatic");
+    set(options, "sz_threadsafe:random_access", "internal options to use random access mode when compiled in");
+    set(options, "sz_threadsafe:rel_err_bound", "the value range relative error bound mode");
+    set(options, "sz_threadsafe:sample_distance", "internal option used to control compression");
+    set(options, "sz_threadsafe:segment_size", "internal option used to control compression. number of points in each segement for pw_relBoundRatio");
+    set(options, "sz_threadsafe:snapshot_cmpr_step", "the frequency of preforming single snapshot based compression in time based compression");
+    set(options, "sz_threadsafe:sol_id", "an internal option use d to control compression");
+    set(options, "sz_threadsafe:sz_mode", "SZ Mode either SZ_BEST_COMPRESSION or SZ_BEST_SPEED");
+    set(options, "sz_threadsafe:user_params", "arguments passed to the application specific mode of SZ. Use in conjunction with sz:app");
+    set(options, "sz_threadsafe:protect_value_range", "should the value range be preserved during compression");
+    return options;
+  }
+
+  sz_params *threadsafe_params;
+
+
+  struct pressio_options get_options_impl() const override {
+    struct pressio_options options;
+    set_type(options, "sz_threadsafe:config_file", pressio_option_charptr_type);
```

Comment:
We might need to remove this without some help from Sheng since we can't apply options directly to a file.